cmd/cli/main.go

```diff
@@ -0,0 +1,375 @@
+						Name:        "restoreToS3Endpoint",
+						Usage:       "Optional ; Set restore endpoint, only when using s3 restore method, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3ENDPOINT",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3Bucket,
+						Required:    false,
+						Name:        "restoreToS3Bucket",
+						Usage:       "Optional ; Set restore bucket, only when using s3 restore method, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3BUCKET",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3Secret,
+						Required:    false,
+						Name:        "restoreToS3Secret",
+						Usage:       "Optional ; Set restore Secret, only when using s3 restore method, expecting secret name containing key value pair with 'username' and 'password' keys, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRET",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3SecretUsernameKey,
+						Required:    false,
+						Value:       "username",
+						Name:        "RestoreToS3SecretUsernameKey",
+						Usage:       "Optional ; Set RestoreToS3SecretUsernameKey, key inside secret, under which S3 username is stored, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRETUSERNAMEKEY",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3SecretPasswordKey,
+						Required:    false,
+						Value:       "password",
+						Name:        "RestoreToS3SecretPasswordKey",
+						Usage:       "Optional ; Set RestoreToS3SecretPasswordKey, key inside secret, under which Restic repo password is stored, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRETPASSWORDKEY",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Namespace,
+						Required:    false,
+						Value:       "default",
+						Aliases:     []string{"n"},
+						Name:        "namespace",
+						Usage:       "Optional ; Set namespace in which You want to execute restore, via cli or via env: ",
+						EnvVars: []string{
+							"NAMESPACE",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Kubeconfig,
+						Required:    false,
+						Value:       "~/.kube/config",
+						Name:        "kubeconfig",
+						Usage:       "Optional ; Set kubeconfig to connect to cluster, via cli or via env:",
+						EnvVars: []string{
+							"KUBECONFIG",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func RunRestore(ctx *cli.Context) error {
+	var restoreName, snapshot string
+	var s3, pvc v1.RestoreMethod
+	logger := cmd.AppLogger(ctx).WithName("cli-restore")
+
+	// avoid name crashes + generate name if naming is not important for user
+	if restore.Cfg.RestoreName == "" {
+		restoreName = "cli-restore-" + restore.RandomStringGenerator(5)
+		log.Println("Creating restore with name: ", restoreName, " in napespace: ", restore.Cfg.Namespace)
```

Comment:
Is it ok to have two types of logs here? there's `log.` and `logger.`

Reply:
yes - that's UX improvement